packages/api2/src/praise/praise.controller.ts

```diff
@@ -51,6 +54,38 @@ export class PraiseController {
     return this.praiseService.findAllPaginated(options);
   }
 
+  @Get('export')
+  @ApiOperation({ summary: 'Export Praises document to json or csv' })
+  @ApiResponse({
+    status: 200,
+    description: 'Praise Export',
+    type: [Praise],
+  })
+  @Permissions(Permission.UsersExport)
+  @ApiParam({ name: 'format', type: String })
+  @ApiParam({ name: 'startDate', type: String })
+  @ApiParam({ name: 'endDate', type: String })
+  @ApiParam({ name: 'periodId', type: String })
+  async export(
+    @Query() options: ExportRequestOptions,
+    @Res({ passthrough: true }) res: Response,
+  ): Promise<Praise[] | undefined> {
+    const praises = await this.praiseService.export(
+      options.format,
+      options.startDate,
+      options.endDate,
+      options.periodId
+    );
+
+    if (options.format === 'json') return praises as Praise[];
```

Comment:
No need to do the `Content-Disposition` etc for json exports?

Reply:
I added content disposition in the controllers, using the response object. It's the way I could make it work.

Reply:
I added the response object to the controller, if the content type is csv I use content disposition.

```
res.set({
      'Content-Type': 'text/csv',
      'Content-Disposition': 'attachment; filename="users.csv"',
    });
res.send(data)
 ``` 

I wasn't 100% sure what is the best way to do it in nestJs, I found that way